plot/plot.app.js

```diff
@@ -164,16 +164,18 @@
                       layout.xaxis = {
                           title: plot.x_axis_label ? plot.x_axis_label : '',
                           automargin: true,
+                          type: config.x_axis_type ? config.x_axis_type : 'auto',
                       };
                       layout.yaxis = {
                           title: plot.y_axis_label ? plot.y_axis_label : '',
                           automargin: true,
+                          type: config.y_axis_type ? config.y_axis_type : 'auto',
                       };
                       return layout;
                   }
                 }
 
-                function formatData(data) {
+                function formatData(data) { // HACK: for the demo data
```

Comment:
Seeing as this function caused an issue earlier, let's change it to use try/catch.
```js
function formatData(data) {
  try {
    var formated_data = parseInt(data.split(' ')[0], 10);
    if (isNaN(formated_data)) {
      return data;
    }
  } catch (e) {
    return data;
  }
}
```

Reply:
This isn't really sufficient, because it will mangle string values which may have numbers in them and that are not really meant to have this special case apply.  A better approach would be to put in a config variable that only executes this behavior if `true`, otherwise this function is either bypassed entirely or is effectively a noop and just returns `data` unmodified.